### ISSUE TITLE
importdescriptors always rescans

### ISSUE THREAD

**Issue by gmart7t2** [bitcoin/bitcoin#31263]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behaviour

Running `importdescriptors` with `timestamp: now` rescans the last 2 hours worth of blocks.

The docs say that '"now" can be specified to bypass scanning' but that doesn't appear to be true:

```
"timestamp": timestamp | "now",    (integer / string, required) Time from which to start rescanning the blockchain for this descriptor, in UNIX epoch time
                                   Use the string "now" to substitute the current synced blockchain time.
                                   "now" can be specified to bypass scanning, for outputs which are known to never have been used, and
                                   0 can be specified to scan the entire blockchain. Blocks up to 2 hours before the earliest timestamp
                                   of all descriptors being imported will be scanned as well as the mempool.
```

I also tried using timestamps several days into the future, but still see about 2 hours worth of blocks being rescanned. I want to be able to import a description without triggering any block rescan.

### Expected behaviour

I expect the behavior to match the docs.

### Steps to reproduce

1. Run importdescriptors with the timestamp set to "now"
2. See in the log that the wallet rescanned some blocks

### How did you obtain Bitcoin Core

Compiled from source

### What version of Bitcoin Core are you using?

v28.0

### Operating system and version

Some kind of Debian

**Comment by furszy** [comment]
> The docs say that '"now" can be specified to bypass scanning' but that doesn't appear to be true

The message also specify that "Blocks up to 2 hours before the earliest timestamp of all descriptors being imported will be scanned as well as the mempool".
This occurs regardless of the provided timestamp. It serves as a grace period to account for possible variations in block times.

**Comment by gmart7t2** [comment]
Even when the earliest timestamp specified is a week in the future it still rescans blocks from 2 hours ago. 

That goes against what the documentation string says.

**Comment by achow101** [comment]
It doesn't make sense for a user to import descriptors with a future timestamp. If the timestamp is greater than "now", then the timestamp for rescanning will be "now".

**Comment by gmart7t2** [comment]
Here's the use case: 

I have a descriptor that I know has never been used. I want to import it. I don't need any rescan, because I know it hasn't been used.

In this case I don't care about variations in block times because 2 hours before "never used" is still "never used".

**Comment by casey** [comment]
Something else that's useful to note is that even a short rescan, on a weak or otherwise loaded machine, can cause RPC calls to start timing out. So avoiding it when unnecessary is useful.

**Comment by furszy** [comment]
> Something else that's useful to note is that even a short rescan, on a weak or otherwise loaded machine, can cause RPC calls to start timing out.

Just a recommendation.
Can make the rescan lighter and faster by enabling `-blockfilterindex`. It will avoid reading and traversing blocks with no matching elements.

**Comment by gmart7t2** [comment]
I have that, and it does make it faster. But I still get RPC timeouts when the machine is loaded, particularly if the wallet already has a lot of descriptors in it. It doesn't only rescan for the one(s) I am adding, but also for all the pre-existing descriptors in the wallet.

**Comment by furszy** [comment]
> I have that, and it does make it faster. But I still get RPC timeouts when the machine is loaded, particularly if the wallet already has a lot of descriptors in it. It doesn't only rescan for the one(s) I am adding, but also for all the pre-existing descriptors in the wallet.

Yes. Thats something I already have in my TO DO list. It requires a rescanning restructuring first.

**Comment by saikiran57** [comment]
Hi, I've provided fix for this issues kindly validate pr and give your valuable comments. https://github.com/bitcoin/bitcoin/pull/31668
